cmd/ci-operator/main.go

```diff
@@ -1380,6 +1385,64 @@ func printExecutionOrder(nodes []*api.StepNode) error {
 	return nil
 }
 
+type stepWithDependencies struct {
+	StepName     string
+	Dependencies []string
+}
+
+func dumpGraph(nodes []*api.StepNode) error {
+	artifactsEnv := os.Getenv(artifactsEnv)
+	// No target to dump to, so lets just skip this
+	if artifactsEnv == "" {
+		return nil
+	}
+
+	var result []stepWithDependencies
+	iterateAllEdges(nodes, sets.String{}, func(n *api.StepNode) {
+		r := stepWithDependencies{StepName: n.Step.Name()}
+		for _, requiment := range n.Step.Requires() {
+			iterateAllEdges(nodes, sets.String{}, func(inner *api.StepNode) {
+				if satisfiedBy(requiment, inner.Step) {
+					r.Dependencies = append(r.Dependencies, inner.Step.Name())
+				}
+			})
+		}
+		result = append(result, r)
+	})
+
+	serialized, err := json.Marshal(result)
+	if err != nil {
+		return fmt.Errorf("failed to marshal: %w", err)
+	}
+
+	dest := filepath.Join(artifactsEnv, "graph.json")
```

Comment:
Can we get a better name? Something that's more descriptive when you eyeball artifacts in the viewer? Or maybe we should claim an artifact subdir? `ci-operator/graph.json` would work for me.

Reply:
Renamed it to `ci-operator-step-graph.json`